runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/operation/SdkOperationExecution.kt

```diff
@@ -260,7 +272,12 @@ internal class HttpAuthHandler<Input, Output>(
         val identityProvider = authScheme.identityProvider(authConfig.identityProviderConfig)
         val identity = identityProvider.resolve(authOption.attributes)
 
-        // FIXME - endpoint resolution should happen here, either we make it explicit or we change implementation to use modifyBeforeSigning and have to stuff identity into the context
+        val resolveEndpointReq = ResolveEndpointRequest(request.context, request.subject.immutableView(), identity)
+        val endpoint = endpointResolver?.resolve(resolveEndpointReq)
+        if (endpoint != null) {
+            coroutineContext.debug<AuthHandler<*, *>> { "resolved endpoint: $endpoint" }
```

Comment:
question: Why not `AuthHandler<Input, Output>`?

Reply:
The `reified` type parameter isn't used for anything other than the `qualified` class name which doesn't retain generic type info anyway ([example](https://pl.kotl.in/Oq49gl2ln))